src/Console/Command/AutoMergeCommand.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DevKit\Console\Command;
+
+use Github\Exception\ExceptionInterface;
+use Github\Exception\RuntimeException;
+use Packagist\Api\Result\Package;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class AutoMergeCommand extends AbstractNeedApplyCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        parent::configure();
+
+        $this
+            ->setName('auto-merge')
+            ->setDescription('Merges branches of repositories if not conflict.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        foreach ($this->configs['projects'] as $name => $projectConfig) {
+            try {
+                $package = $this->packagistClient->get(static::PACKAGIST_GROUP.'/'.$name);
+                $this->io->title($package->getName());
+                $this->mergeBranches($package, $projectConfig);
+            } catch (ExceptionInterface $e) {
+                $this->io->error('Failed with message: '.$e->getMessage());
+            }
+        }
+
+        return 0;
+    }
+
+    private function mergeBranches(Package $package, array $projectConfig)
+    {
+        if (!$this->apply || !array_key_exists('branches', $projectConfig)) {
+            return;
+        }
+
+        $repositoryName = $this->getRepositoryName($package);
+        $branches = array_reverse(array_keys($projectConfig['branches']));
+
+        while (($head = current($branches))) {
+            $base = next($branches);
+            if (false === $base) {
```

Comment:
> why the reverse

Because we merge the oldest branch to the newest.

> why not use foreach

Proper. We avoid to have to create an index variable to check the next one.

> and why check for false

`next` return `false` if there is nothing after the current element.

See: http://php.net/manual/fr/function.next.php